src/Workspaces/Core/Portable/CodeActions/Operations/CodeActionOperation.cs

```diff
@@ -25,10 +27,16 @@ public virtual void Apply(Workspace workspace, CancellationToken cancellationTok
         {
         }
 
-        internal virtual bool TryApply(Workspace workspace, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        /// <summary>
+        /// Called by the host environment to apply the effect of the operation.
+        /// This method is guaranteed to be called on the UI thread.
```

Comment:
> (We'd have to keep some compat shim in for the sync method, but I wouldn't object to using @sharwell's backdoor we have in the Workspaces layer, since this would be a classic "we have an API we regret that is async-over-sync" and it's really for back compat only.)

I'm fine with this.  I just have no idea what that backdoor is :)